contribute/design/1.x/cstor/pool/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,135 @@
+		- [User Stories](#user-stories)
+		- [Current Implementation](#current-implementation)
+		- [Proposed Implementations](#proposed-implementations)
+		- [Steps to perform user stories](#steps-to-perform-user-stories)
+			- [Pool Expansion Steps:](#pool-expansion-steps)
+		- [Low Level Design](#low-level-design)
+			- [Work Flow](#work-flow)
+	- [Drawbacks](#drawbacks)
+	- [Alternatives](#alternatives)
+	- [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+- This proposal includes design details of the pool expansion when underlying disks were expanded.
+
+## Motivation
+
+There were cases where OpenEBS users expand the pool by expanding underlying disks(which were used by the pool) when space on the pool got exhausted.
+
+### Goals
+
+- Expand cStor pool when the underlying disk got expanded.
+
+### Non-Goals
+
+- If the user dedicated the first part of the partitioned disk for cStor and second part for third-party then pool expansion can't be performed automatically until and unless the user extends the first partition.
+
+## Proposal
+
+### User Stories
+
+As an OpenEBS user, I should be able to expand the pool when the underlying disk got expanded.
+
+### Current Implementation
+
+- When the user creates the CSPC, the CSPC controller which is watching for CSPC will get create requests and creates cstorpoolinstances and cstorpooldeployments. CstorPoolInstance is responsible for holding the blockdevice information and other pool configurations whereas cStor pool deployment have 3 containers cstor-pool-mgmt, cstor-pool and m-exporter as the name conveys cstor-pool container creates pool(ZFS file system) on top of the blockdevices, whereas cstor-pool-mgmt sidecar container is responsible for managing these pool. If any changes required as of day 2 operations on the pool then pool configurations will be updated by adding BlockDevices, adding RaidGroup and replacing blockdevices cstor-pool-mgmt looks for any pending pool operations for CStorPoolInstance then performs pending operation on the pool.
+
+Below is high-level info to explain the Day2 operations that are currently supported.
+
+- Pool Expansion When BlockDevice/raid group added:
+	cstor-pool-mgmt will iterate over all the BlockDevices that exist in CStorPoolInstance resource and verify whether the corresponding BlockDevice/raid group is part of the pool. If that BlockDevice/raid group is not part of the pool then pool expansion will be triggered on newly added BlockDevice/raid group.
+
+- BlockDevice Replacement In Pool:
+    cstor-pool manager will iterate over all the BlockDevices present in CstorPoolInstance and verifies whether the current blockdevice is for BlockDevice replacement request[How cstor-pool-mgmt can know? We are maintaining annotation called openebs.io/predecessor: <replaced_blockdevice_name> on the claim of that blockdevice]? If requested BlockDevice is valid for replacement then the pool manager will perform BlockDevice replacement operation on the pool.
```

Comment:
`cstor-pool-mgmt` and `cstor-pool manager`. are they same?

Is the annotation `openebs.io/predecessor: <replaced_blockdevice_name>` provided by the user.

Reply:
> cstor-pool-mgmt and cstor-pool manager. are they same?

Yes, both are the same I will update this doc with cstor-pool-manager as we are building container as cstor-pool-manager.

> Is the annotation openebs.io/predecessor: <replaced_blockdevice_name> provided by the user.

It will add by admission server.